### PR TITLE
The modularized typings definitions for iscroll package

### PR DESCRIPTION
**Pull request by acrazing**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <http://iscrolljs.com/>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

### CODE REVIEW

**Comment by acrazing** [comment]
The travis-ci error is generated by `types-publisher`, I created a PR to fix it: <https://github.com/Microsoft/types-publisher/pull/296>

**Comment by andy-ms** [comment]
Why did you delete `v4`?

**Comment by acrazing** [comment]
@andy-ms 

If user need to use `v4`, just use `yarn add "@types/iscroll@4.2.29"` to invoke it. This version does not support `v4` because there is no document.

The old version use global export, it's useless. <https://github.com/DefinitelyTyped/DefinitelyTyped/pull/14591/files#diff-20f2a108c9f72059c3f607f091a732d1L35>

And too much code error, for example: <https://github.com/DefinitelyTyped/DefinitelyTyped/pull/14591/files#diff-20f2a108c9f72059c3f607f091a732d1L38>

So I removed it.

**Comment by minestarks** [comment]
@acrazing Are the tests supposed to be fixed? Could you re-run the checks before I can merge? Thanks.

File: iscroll/tslint.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "extends": "../tslint.json",
+  "rules": {
+    "interface-name": false
```
Comment (minestarks) [inline iscroll/tslint.json]: Why was it necessary to disable this rule?

File: iscroll/tsconfig.json [hunk 2]
```diff
@@ -1,23 +1,23 @@
...
\ No newline at end of file
+  "compilerOptions": {
+    "module": "commonjs",
+    "lib": [
+      "es6",
+      "dom"
+    ],
+    "noImplicitAny": true,
+    "noImplicitThis": true,
+    "strictNullChecks": false,
```
Comment (minestarks) [inline iscroll/tsconfig.json]: Please turn ```strictNullChecks``` on

File: iscroll/v4/index.d.ts [hunk 3]
```diff
@@ -1,76 +0,0 @@
-// Type definitions for iScroll 4.2
```
Comment (minestarks) [inline iscroll/v4/index.d.ts]: It's valid to leave v4 in. This way it can receive bugfixes. Consider un-deleting please.

**Review by minestarks — CHANGES_REQUESTED** [review]

**Pull request closed without merging**